src/FSharp.Formatting.ApiDocs/GenerateMarkdown.fs

```diff
@@ -0,0 +1,487 @@
+  
+      // Show the summary (and sectioned docs without any members)
+      p [ embed entity.Comment.Summary ]
+
+      // Show the remarks etc.
+      match entity.Comment.Remarks with
+      | Some r ->
+          p [embed r]
+      | None -> ()
+
+      for note in entity.Comment.Notes do 
+          ``#####`` [!! "Note"]
+          p [embed note]
+
+      for example in entity.Comment.Examples do 
+          ``#####`` [!! "Example"]
+          p [embed example]
+
+      if (byCategory.Length > 1) then
+        // If there is more than 1 category in the type, generate TOC 
+        ``###`` [!! "Table of contents"]
+        ul [
+          for (index, _, name) in byCategory do
+            [p [link [!! (sprintf "#section%d" index)] (name)]]
+        ]
+ 
+      //<!-- Render nested types and modules, if there are any -->
+  
+      let nestedEntities =
+         entity.NestedEntities
+         |> List.filter (fun e -> not e.IsObsolete)
+  
+      if (nestedEntities.Length > 0) then
+        
+        ``###`` [!! (if nestedEntities |> List.forall (fun e -> not e.IsTypeDefinition)  then "Nested modules"
+                     elif nestedEntities |> List.forall (fun e -> e.IsTypeDefinition) then "Types"
+                     else "Types and nested modules")]
+        yield! renderEntities nestedEntities
+
+      for (index, ms, name) in byCategory do
+        // Iterate over all the categories and print members. If there are more than one
+        // categories, print the category heading (as <h2>) and add XML comment from the type
+        // that is related to this specific category.
+        if (byCategory.Length > 1) then ``##`` [!! name]
+
+        yield! renderMembers "Functions and values" "Function or value" (ms |> List.filter (fun m -> m.Kind = ApiDocMemberKind.ValueOrFunction))
+        yield! renderMembers "Type extensions" "Type extension" (ms |> List.filter (fun m -> m.Kind = ApiDocMemberKind.TypeExtension))
+        yield! renderMembers "Active patterns" "Active pattern" (ms |> List.filter (fun m -> m.Kind = ApiDocMemberKind.ActivePattern))
+        yield! renderMembers "Union cases" "Union case" (ms |> List.filter (fun m -> m.Kind = ApiDocMemberKind.UnionCase))
+        yield! renderMembers "Record fields" "Record Field" (ms |> List.filter (fun m -> m.Kind = ApiDocMemberKind.RecordField))
+        yield! renderMembers "Static parameters" "Static parameters" (ms |> List.filter (fun m -> m.Kind = ApiDocMemberKind.StaticParameter))
+        yield! renderMembers "Constructors" "Constructor" (ms |> List.filter (fun m -> m.Kind = ApiDocMemberKind.Constructor))
+        yield! renderMembers "Instance members" "Instance member" (ms |> List.filter (fun m -> m.Kind = ApiDocMemberKind.InstanceMember))
+        yield! renderMembers "Static members" "Static member" (ms |> List.filter (fun m -> m.Kind = ApiDocMemberKind.StaticMember))
+    ]
+
+  let categoriseEntities (nsIndex: int, ns: ApiDocNamespace, suppress) =
```

Comment:
I think this code can be shared between Html and Markdown genreators?

Reply:
Indeed, it needs a bit of refactoring.

Reply:
Good work on resolving this, thank you